src/v/reflection/type_traits.h

```diff
@@ -33,6 +34,14 @@ template<typename T, template<typename...> class C>
 inline constexpr bool is_specialization_of_v
   = is_specialization_of<T, C>::value;
 
+template<class T, template<class, size_t> class C>
```

Comment:
serde.h depends on reflection/type_traits.h and defines the same constexprs so you may remove a definition there